### PR TITLE
[server] Fix clientIp extraction for analytics

### PR DESCRIPTION
**Pull request by geropl**
## Description
Fixes extraction of (masked) client IP for websocket connections.

Turns this output
image
into this
image

## Related Issue(s)

Follow-up for: https://github.com/gitpod-io/gitpod/pull/14978

## How to test
- turn on debug logging with: `gpctl debug log server`
- visit the [preview env's dashboard](https://gpl-analytics.preview.gitpod-dev.com/workspaces)
- check server logs, and notice these lines:
image

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@geropl: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/server/src/websocket/websocket-connection-manager.ts
Comment (easyCZ) [inline components/server/src/websocket/websocket-connection-manager.ts:249]: Do you want to keep this in the PR or wanted to remove it before merging?

**Review by easyCZ — APPROVED** [review]
LGTM

/hold if you wanted to remove the debug log

File: components/server/src/websocket/websocket-connection-manager.ts
Comment (geropl) [reply components/server/src/websocket/websocket-connection-manager.ts:249]: I wanted this to hit prod so we can turn it on if we want; and remove it in a subsequent PR. :+1:

**Pull request merged**
